pkg/webhooks/mutator/virt-launcher-mutator.go

```diff
@@ -0,0 +1,234 @@
+package mutator
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	"github.com/go-logr/logr"
+	"gomodules.xyz/jsonpatch/v2"
+	admissionv1 "k8s.io/api/admission/v1"
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/kubevirt/hyperconverged-cluster-operator/api/v1beta1"
+)
+
+var _ admission.Handler = &VirtLauncherMutator{}
+
+const (
+	cpuLimitToRequestRatioAnnotation    = "kubevirt.io/cpu-limit-to-request-ratio"
+	memoryLimitToRequestRatioAnnotation = "kubevirt.io/memory-limit-to-request-ratio"
+
+	launcherMutatorStr = "virtLauncherMutator"
+)
+
+type VirtLauncherMutator struct {
+	cli          client.Client
+	hcoNamespace string
+	decoder      *admission.Decoder
+	logger       logr.Logger
+}
+
+func NewVirtLauncherMutator(cli client.Client, hcoNamespace string) *VirtLauncherMutator {
+	return &VirtLauncherMutator{
+		cli:          cli,
+		hcoNamespace: hcoNamespace,
+		logger:       log.Log.WithName("virt-launcher mutator"),
+	}
+}
+
+func (m *VirtLauncherMutator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	m.logInfo("Starting virt-launcher mutator handling")
+
+	if req.Operation != admissionv1.Create {
+		m.logInfo("not a pod creation - ignoring")
```

Comment:
if we got here, i think it is an error in the webhook definition, since it is defined to forward the request to the webhook handler only on pods creation:
https://github.com/kubevirt/hyperconverged-cluster-operator/pull/2206/files#diff-a9655738a5e2e547cd22e5e3a3b56329022658b0ae27e94c49f54c18d12ce362R3853-R3861

Reply:
Yeah that's right. I just prefer to also have it here as a second line of defense. It also serves as "documentation", since the configuration is located in a different file that would be hard to find without context. In addition, the current namespace mutator does the same thing with deletions.